h1b/src/nvcounter/capsule.rs

```diff
@@ -0,0 +1,293 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use ::kernel::ReturnCode;
+use super::internal::*;
+use super::traits::{Client,NvCounter};
+use crate::hil;
+
+/// NvCounter implementation using flash memory.
+
+// Uses two pages of flash, flipping two bits to zero in each page at a time.
+// The low page is struck or erased for every increment operation. When the low
+// page wraps around, the high page is struck once, the low page is erased, then
+// the high page is struck again. This allows the counter to resume if the
+// increment operation is interrupted, providing atomicity.
+//
+// initialize() sequence:
+//   A1. Erase low
```

Comment:
I'd suggest using names rather than symbols. E.g., call then sequence EraseLow, EraseHigh, WriteLow, WriteHigh, EraseLow, WriteHigh2. Referring to C2, A2, in the code is confusing.

Reply:
I updated them. I kept the (sequence, step number) pattern but gave the sequences more meaningful names.